reference/nats-protocol/nats-protocol/README.md

```diff
@@ -162,15 +164,54 @@ where:
 
 To publish the ASCII string message payload "Hello NATS!" to subject FOO:
 
-`PUB FOO 11\r\nHello NATS!\r`
+`PUB FOO 11␍␊Hello NATS!␍␊`
 
 To publish a request message "Knock Knock" to subject FRONT.DOOR with reply subject INBOX.22:
 
-`PUB FRONT.DOOR INBOX.22 11\r\nKnock Knock\r`
+`PUB FRONT.DOOR INBOX.22 11␍␊Knock Knock␍␊`
 
 To publish an empty message to subject NOTIFY:
 
-`PUB NOTIFY 0\r\n\r`
+`PUB NOTIFY 0␍␊␍␊`
+
+## HPUB
+
+### Description
+
+The `HPUB` message is the same as `PUB` but extends the message payload to include NATS headers. Note that the payload itself is optional. To omit the payload, set the total message size equal the size of the headers. Note that the trailing CR+LF is still required.
+
+NATS headers are similar, in structure and semantics, to HTTP headers as `name: value` pairs including supporting multi-value headers. Headers can be mixed case and NATS will preserve case between message publisher and message receiver(s).  See also [ADR-4 NATS Message Headers](https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-4.md).
+
+### Syntax
+
+`HPUB <subject> [reply-to] <#header bytes> <#total bytes>␍␊[headers]␍␊␍␊[payload]␍␊`
+
+where:
+
+* `subject`: The destination subject to publish to
+* `reply-to`: The optional reply inbox subject that subscribers can use to send a response back to the publisher/requestor
```

Comment:
Since an *inbox* subject is not technically required (in the `_INBOX.>` sense of the word).. I would suggest the word be removed since it may be misinterpreted as a requirement.

If you agree, there are a few other places where "inbox" it used that should be updated.

Reply:
I agree.  As long as we've opened up the patient, we can do some additional nips and tucks.